### ISSUE TITLE
[DISCOVERY] Mapbox Accessibility

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#8535]
## Problem Statement
How might we best resolve the accessibility issues found with Mapbox?

## Background
A number of accessibility defects have been identified within the Mapbox feature on Facility Locator. Our ability to resolve them may be impacted by Mapbox's functionality and/or library and version currently in use. 

## Information from Mapbox: 

> After taking a look at the map at https://www.va.gov/find-locations/, it looks like you are currently working with Leaflet. We have an accessibility plugin available, but it must specifically be used with Mapbox GL JS. With this plugin, you can use the addControl function to add an instance of `MapboxAccessibility` to the map, which allows you to define all of the layers and properties in those layers that should be used for voiceover purposes. 
The plugin adds an invisible HTML button element on top of each feature with an aria-label property based on what is defined as the `accessibleLabelProperty`. This makes it so assistive technology interprets the symbol as text that is read aloud as the user tabs over the feature. 

**To provide instructions that indicate interactive map keyboard shortcuts:** Add an aria-label to explain the keys to be used to access the map ("Use the +/- keys to zoom in and out on the map. Use the arrow keys to pan.") 

**To ensure keyboard users are able to get to the map:** The map sits within an element on the page that will be accessible to users working with Voiceover. Anything additional you would like to add to be announced when you tab into the map element could be added with an aria-label. 

**Ensure there is an indication of the keyboard shortcuts provided when the mouse is used:** could use DOM events to add the label if the mouse is being used on the page. 

**Ensure speech input users are able to interact with the map controls by speaking the visible text labels.**  Mapbox doesn't have any built-in speech recognition functionality, but you should be able to filter by the text label (or letter based on the implementation at https://www.va.gov/find-locations/) announced by applying the same logic that you do on click to place the appropriate panel in focus. 

## Goal
The goal of this effort is to determine the path forward for resolution of Mapbox feature on Facility Locator. 
- Can the issues be resolved with current implementation?
- Can the issues be resolved with upgrade to GL JS and if so, does the impact for users balance the effort? 
- Is blocking the map from assistive tech a viable, reasonable and/or acceptable option based on findings? (Or add "view the map" user action)

## Objectives or Key Results this is meant to further
- Objective 3: Deliver a usable and accessible experience

## Considerations
- Based on above from Mapbox engineers, identify which accessibility issues (and at which level of defect) can be resolved with current implementation
- Identify which accessibility issues (and at which level of defect) could be resolved with change to Mapbox GL JS
- Evaluate what features/functionality would be gained and lost by move to Mapbox GL JS from Leaflet
- Sitewide impact caused by changing to Mapbox GL JS
- Contract/pricing between Mapbox GL JS and Leaflet
- Lift to move to Mapbox GL JS
- Do the answers to these questions allow us to make a decision or will we need a prototype to investigate/test with accessibility team? 

## Acceptance Criteria
- [ ] Determine approach for resolving accessibility findings
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by brangi** [comment]
@mmiddaugh  Experimental PR to explore the changes - https://github.com/department-of-veterans-affairs/vets-website/pull/14511

I will leave the PR on draft mode if we want to re-visit more implementation later

**Comment by brangi** [comment]
Discussion around the topic https://dsva.slack.com/archives/C0FQSS30V/p1602185477315900

**Comment by brangi** [comment]
@mmiddaugh I will close this as the outcome is somehow clear.
